mantidproject/autoreduce
# Log viewing

Fixes #166 

With this, logging is saved to string buffers in memory in `PostProcessAdmin` as well as the usual log files - both the admin logging itself and the script's logging, separately. These strings are then sent back to the webapp, which stores them in the database. There is now a link on the run summary page to "view logs", which brings up a pop-up with these logs shown. You can also click the error notification at the top to do the same.

This also fixes a few misc. things, like deleting troublesome migration files, and fixing a small bug involving reruns and their variables.

With reference to
>Send the whole log back as part of the message (bad as don't want to send too much data around as messages)
in the issue; messages can happily send around a few KBs of text, and orders of magnitude more data per message than we'll ever get to with text logs. So this isn't an issue.

To test:
* Run a reduction run that succeeds. Check that the 'preview logs' link displays a pop-up with reduction and admin logs (merged into one).
* Run a reduction run that fails. Check that the error message links to the log preview.

**NB**: like some of the other PRs, going from an old database to this one requires a Django `migrate` command to add new fields to the database. The production server currently (13/09/16) has a database up to date with this PR, which is backwards compatible with master, so **nothing needs to be done**. However, because before this PR some migration files were committed to the repo, Django's `migrate` will be mistaken about what needs to be migrated, so don't trust it.
